contracts/Example2.sol

```diff
@@ -2,10 +2,18 @@ pragma solidity ^0.8.4;
 
 contract Example2 {
     uint public counter;
- 
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+
+    function incrementBy(uint[] calldata arr) public {
+        uint array_len = arr.length;
```

Comment:
The general naming convention in solidity is camel casing, similar to one used for function naming here